src/main/java/com/librato/disco/Node.java

```diff
@@ -5,26 +5,29 @@
 /**
  * Represents information about a particular discovered node
  */
-public class Node {
+public class Node<T> {
     public final String host;
     public final int port;
+    public final T payload;
 
-    public Node(String host, int port) {
+    public Node(String host, int port, T payload) {
```

Comment:
I think we have to allow the payload to be null *at least* for backwards compatibility.